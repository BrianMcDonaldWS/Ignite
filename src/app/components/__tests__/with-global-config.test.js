import React from 'react';
import withGlobalConfig from '../WithGlobalConfig';
import renderToJson from './utils/render-to-json';

describe('withGlobalConfig', () => {
  let component;

  beforeAll(() => {
    window.configuration = {
      markdown: [
        ['index.md', () => <div> Foo </div>],
        ['page.md', () => <div> Foo </div>]
      ],
      plugins: [['somePlugin', () => <div> Plugin </div>]]
    };

    component = () => {};
  });

  test('renders component with configuration', () => {
    const ComponentWithConfig = withGlobalConfig(component);
    expect(renderToJson(<ComponentWithConfig />)).toMatchSnapshot();
  });

  test('renders blog hero', () => {
    const ComponentWithConfig = withGlobalConfig(component);
    expect(renderToJson(<ComponentWithConfig />)).toMatchSnapshot();
    window.configuration.setBlogHero('http://some.link');
    expect(renderToJson(<ComponentWithConfig />)).toMatchSnapshot();
  });

  test('sets first page', () => {
    const ComponentWithConfig = withGlobalConfig(component);
    expect(renderToJson(<ComponentWithConfig />)).toMatchSnapshot();
    window.configuration.setFirstLink('index.md', 'page.md');
    expect(renderToJson(<ComponentWithConfig />)).toMatchSnapshot();
  });
});
