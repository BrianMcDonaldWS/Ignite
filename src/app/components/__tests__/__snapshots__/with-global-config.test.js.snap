// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withGlobalConfig renders blog hero 1`] = `
<component
  blogHero={null}
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig renders blog hero 2`] = `
<component
  blogHero="http://some.link"
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig renders component with configuration 1`] = `
<component
  blogHero={null}
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig sets first page 1`] = `
<component
  blogHero="http://some.link"
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig sets first page 2`] = `
<component
  blogHero="http://some.link"
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
        undefined,
        "page.md",
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig sets search index 1`] = `
<component
  blogHero="http://some.link"
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
        undefined,
        "page.md",
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
/>
`;

exports[`withGlobalConfig sets search index 2`] = `
<component
  blogHero="http://some.link"
  markdown={
    Array [
      Array [
        "index.md",
        [Function],
        undefined,
        "page.md",
      ],
      Array [
        "page.md",
        [Function],
      ],
    ]
  }
  plugins={
    Array [
      Array [
        "somePlugin",
        [Function],
      ],
    ]
  }
  searchIndex={
    Object {
      "page.md": "all the markdown text for page.md",
    }
  }
/>
`;
